src/cpp/rtps/writer/LivelinessManager.cpp

```diff
@@ -137,34 +154,53 @@ bool LivelinessManager::assert_liveliness(
         {
             if (w.kind == wit->kind)
             {
-                if (w.alive == false)
+                if (callback_ != nullptr)
```

Comment:
This code is repeated in several places. I propose to include methods inside LivelinessData struct to handle the status transitions. They should either receive the callback as a parameter or have out parameters to fill the increments and return a boolean indicating if the callback should be performed.

Reply:
I've moved some common code into a function